cmd/register/register_util.go

```diff
@@ -414,27 +444,70 @@ func getJSONSpec(message proto.Message) string {
 	return jsonSpec
 }
 
-func getFlyteTestManifest() ([]FlyteSnack, string, error) {
+func getFlyteTestManifest(org, repository string) ([]FlyteSnack, string, error) {
 	c := github.NewClient(nil)
 	opt := &github.ListOptions{Page: 1, PerPage: 1}
-	releases, _, err := c.Repositories.ListReleases(context.Background(), githubOrg, githubRepository, opt)
+	releases, _, err := c.Repositories.ListReleases(context.Background(), org, repository, opt)
 	if err != nil {
 		return nil, "", err
 	}
-	response, err := http.Get(fmt.Sprintf(flyteManifest, *releases[0].TagName))
-	if err != nil {
-		return nil, "", err
+	if len(releases) > 0 {
+		response, err := http.Get(fmt.Sprintf(flyteManifest, *releases[0].TagName))
+		if err != nil {
+			return nil, "", err
+		}
+		defer response.Body.Close()
+
+		data, err := ioutil.ReadAll(response.Body)
+		if err != nil {
+			return nil, "", err
+		}
+
+		err = json.Unmarshal(data, &FlyteSnacksRelease)
+		if err != nil {
+			return nil, "", err
+		}
+		return FlyteSnacksRelease, *releases[0].TagName, nil
 	}
-	defer response.Body.Close()
+	return nil, "", fmt.Errorf("Repository doesn't have any release")
+}
+
+func getAdditionalDistributionLoc(remoteLocation, identifier string) storage.DataReference {
+	return storage.DataReference(fmt.Sprintf("%v/%v.tar.gz", remoteLocation, identifier))
+}
 
-	data, err := ioutil.ReadAll(response.Body)
+func uploadFastRegisterArtifact(ctx context.Context, file string, storageClient storage.ComposedProtobufStore) error {
+	var dataRefReaderCloser io.ReadCloser
+	fullRemotePath := getAdditionalDistributionLoc(rconfig.DefaultFilesConfig.AdditionalDistributionDir, rconfig.DefaultFilesConfig.Version)
+	raw, err := json.Marshal(file)
 	if err != nil {
-		return nil, "", err
+		return err
 	}
+	dataRefReaderCloser, err = os.Open(file)
+	if err != nil {
+		return err
+	}
+	dataRefReaderCloser, err = gzip.NewReader(dataRefReaderCloser)
+	if err != nil {
+		return err
+	}
+	if err := storageClient.WriteRaw(ctx, fullRemotePath, int64(len(raw)), storage.Options{}, dataRefReaderCloser); err != nil {
```

Comment:
Does this require separate permissions if its going to write to S3 . What is the expectation of the policy of the passed in bucket  .Can you elaborate that and add it in the docs

Reply:
ok, flyte use standard configuration across all go component. Storage config works the same as it works in another component. I didn't add permission until now. I can leave it open for more discussion 

CC: @kumare3 @EngHabu